subsys/storage/stream/stream_flash.c

```diff
@@ -115,7 +115,7 @@ int stream_flash_buffered_write(struct stream_flash_ctx *ctx, const u8_t *data,
 	size_t fill_length;
 	u8_t filler;
 
-	if (!ctx || !data) {
+	if (ctx == NULL || (data == NULL && len != 0)) {
```

Comment:
Shouldn't NULL be valid address of source  for any copy operation?

Reply:
Hm, true. It will not do any harm at least.

Reply:
So... are you going to let it do copy from NULL address? Maybe check if len is non-0 is enough?

Reply:
Then we can simply remove the `!data` check, we shouldn't fail for 0 length writes anyway IMO.

Reply:
Please check new version

Reply:
Why have you resigned from checking `len`?  I actually agree with you on checking `len` against 0, I have just trouble understanding whether it should be EINVAL or just instant success. Some arch's opinion would be handy...

Reply:
When flushing the `len` will typically be 0, so we need to allow this. In my first proposal, I only allowed `data == NULL` when `len == 0`, but since we now treat `data == 0` as a valid source, I don't see any reason to check the `len` field anymore as 0 is always a valid value (as it used to be, it was just used to conditionally allow `data` to be `0`).

Reply:
Yeah, but we could bail out from doing any work when len is found to be 0.
I am passing the review, as this is not important.

Reply:
When `flush` is set there is work to do even with `len == 0`